### DISCUSSION TITLE
Frigate on aws ec2 instance

### DISCUSSION THREAD

**Discussion by sgrossiP** [blakeblackshear/frigate#7536] [Ask A Question]
Hello, anyone has tried to run frigate in an ec2 instance? No recognition obviously.
im about to buy an instance any recommendation or experiencie will be useful

**Comment by hagak** [comment]
What is the goal here?  Cause in this case from a cost/performance standpoint EC2 will never be able to compete with a local system.  You will have to have a pretty constant upload bandwidth usage by your local cameras streaming to AWS, which even at low res and fps this can quickly add up once you get 10 or so cameras.  Also the latency is usually a thing people want as low as possible with an object detection system if using it for automation and AWS will mostly add ~40-50ms on top of that.  Maybe less if you have a really good network connection and close to an AWS Region.

Also if you go this route you really need to pay attention to security, Frigate does not have builtin auth so you will need to implement that in front of it.  I would strongly recommend having a VPN gateway to your VPC that your cameras stream over since most cameras have little to no security on them.

**Reply by sgrossiP** [reply]
well i have several cameras in different locations. in order to avoid some time without electricity i was wondering if amazon will not be so much expensive
yes my frigate its going to be exposed to the network but using a nginx with cors, a custom domain and some kind of validation in the http file of my frigate will be enough
btw my ec2 instance is running but does not have frigate installed yet but i can provide you ssh access for testing if you require, and anyone else interested in
